JASP-Engine/JASP/R/contingencytables.R

```diff
@@ -15,1778 +15,1288 @@
+    crossTabKendallTau$position <- 6
+    
+    # Add columns to table
+    .crossTabLayersColumns(crossTabKendallTau, analysis)
+    crossTabKendallTau$addColumnInfo(name = "value[kTauB]", title = "Kendall's Tau-b ", 
+                                     type = "number")
+    crossTabKendallTau$addColumnInfo(name = "statistic[kTauB]", title = "Z", 
+                                     type = "number", format = "dp:3")
+    crossTabKendallTau$addColumnInfo(name = "p[kTauB]", title = "p", type = "pvalue")
+    if (options$VovkSellkeMPR) 
+      crossTabKendallTau$addColumnInfo(name = "MPR[kTauB]", title = "VS-MPR\u002A", 
+                                       type = "number")
+
+    analysisContainer[["crossTabKendallTau"]] <- crossTabKendallTau
+    analysis <- as.list(analysis)
+    # Compute/get Group List
+    groupList <- .crossTabComputeGroups(dataset, options, analysisContainer, analysis, ready)
+    
+    res <- try(.crossTabKendallsTauRows(analysisContainer, analysis$rows, groupList, options, ready))
+    .crossTabSetErrorOrFill(res, crossTabKendallTau)
+  }
+}
 
-			if (status$ready == FALSE) {
+# Table functions
+.crossTabLayersColumns <- function(table, analysis) {
+  if ((length(analysis)) >= 3)
+    for (j in (length(analysis)):3)
+      table$addColumnInfo(name = analysis[[j]], type = "string", combine = TRUE)
+}
 
-				# do nothing
+.crossTabMainOvertitle <- function(dataset, options, table, analysis, counts.fp) {
+  
+  lvls <- c()
+  overTitle <- unlist(analysis$columns)
+  
+  if(analysis$columns == "") {
+    lvls <- c("a", "b")
+    for (column.name in lvls) {
+      private.name <- paste0(column.name,"[counts]")
+      
+      if (counts.fp || options$countsExpected || options$percentagesRow || 
+          options$percentagesColumn || options$percentagesTotal ) 
+        table$addColumnInfo(name = private.name, title = ".", 
+                            overtitle = ".", type = "number", format = "sf:4;dp:2")
+      else 
+        table$addColumnInfo(name = private.name, title = ".", 
+                            overtitle = ".", type = "integer")
+      
+      if (options$countsExpected) { 
+        private.name <- paste0(column.name,"[expected]")
+        table$addColumnInfo(name = private.name, title = ".", 
+                            type = "number", format = "sf:4;dp:2")
+      }
+      if (options$percentagesRow) {
+        private.name <- paste0(column.name,"[row.proportions]")
+        table$addColumnInfo(name = private.name, title = ".", 
+                            type = "number", format = "dp:1;pc")
+      }
+      if (options$percentagesColumn) {
+        private.name <- paste0(column.name,"[col.proportions]")
+        table$addColumnInfo(name = private.name, title = ".", 
+                            type = "number", format = "dp:1;pc")
+      }
+      if (options$percentagesTotal) {
+        private.name <- paste0(column.name,"[proportions]")
+        table$a
```

Comment:
For all footnotes: if they pertain to a specific row, can you ensure that a symbol is added to that row and the footnote itself refers to that row?
Right now they are added as general footnotes and users might not understand they refer to a row and not the entire table.